lms/djangoapps/courseware/views/views.py

```diff
@@ -1631,6 +1631,9 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
             u"Rendering of the xblock view '{}' is not supported.".format(bleach.clean(requested_view, strip=True))
         )
 
+    staff_access = has_access(request.user, 'staff', course_key)
+    _course_masquerade, request.user = setup_masquerade(request, course_key, staff_access)
```

Comment:
Allows viewing the shift dates CTAs when masquerading. All other spots where request and request.user are used in this function should be using the masqueraded learner rather than the staff user.